### ISSUE TITLE
Q2admin look for config files in root ?

### ISSUE THREAD

**Issue by PowaBanga** [tastyspleen/q2admin-tsmod#4]
Hi,
When i compil and install q2pro with q2admin in root (on linux), q2admin look for config files in root, at the place of look for them into user's local folder.

is there a way to set the config path where q2admin will look for config files ?

**Comment by QwazyWabbitWOS** [comment]
Place the q2admin dll and all its config files in the same folder as the mod you want to run. Never compile as root. Always compile as ordinary user. A copy of the dll and the config files must exist in each mod folder for each mod you intend to run. 

In each case the q2admin dll will be named gamei386.so and the game mod will be named gamei386.real.so. This is called cascading the dlls.

**Comment by PowaBanga** [comment]
yeah, it is that i was saying ^^

It is not usefull if we want to create any linux packages with Q2admin in, if it do not look for config files in user path before ^^

**Comment by spatulasnout** [comment]
Hi, I'm not sure I'm understanding.

For me, Q2admin looks for config files in:

`./$GAME/q2admin.txt`

Where $GAME is the Q2 mod directory, like baseq2, ra2, ctf, tourney, etc.

**Comment by QwazyWabbitWOS** [comment]
You have the source. You are free to modify it and distribute it as part of your package distribution but Quake 2 itself does not support running as a per-user process. You will have to create your own package installation and modify the core code plus all the DLLs for all the mods and many mods are no longer available in source form.

**Comment by QwazyWabbitWOS** [comment]
Can you please explain where your are putting the q2pro executable and the DLLs and the mods you intend to run? A diagram of the directory tree would be helpful.

**Comment by PowaBanga** [comment]
you are right @QwazyWabbitWOS , i fact if is for create a package for dday mod.
https://github.com/PowaBanga/DDaynormandyFPS

The game should be for exemple here : 
```
/opt/game/ddaynormandy
/opt/game/ddaynormandy/dday/gamei386.real.so
/opt/game/ddaynormandy/dday/gamei386.so
```

and the config files, here : 
```
$HOME/.ddaynormandy
$HOME/.ddaynormandy/dday/config.cfg
$HOME/.ddaynormandy/dday/q2admin.txt
$HOME/.ddaynormandy/dday/q2admin.txt
$HOME/.ddaynormandy/dday/q2daminlog.txt
$HOME/.ddaynormandy/dday/q2adminlrcon.txt
$HOME/.ddaynormandy/dday/q2adminflood.txt
$HOME/.ddaynormandy/dday/q2adminspawn.txt
...
```
ore, why not : 
```
$HOME/.ddaynormandy/dday/q2admin/q2admin.txt
...
```

The main objective is to create a mecanism where i do not have to modify your source to change that when i compil my release of dday. For exemple by `make --config-path=$HOME/.ddaynormandy`

But your are right, maybe can i try to change that myself in your source and do a pull request...

**Comment by QwazyWabbitWOS** [comment]
I've never installed Quake2 or its mods in /opt/ but I suppose there might be times when it's appropriate. I don't manage Linux servers and in all my Linux instances I installed my game in ~/quake2/ and I don't make them hidden. The Q2 engine opens its own file system at initialization, rooted in the process's startup directory and it looks there first for configuration files, then in whatever game folder you specified in the command line. I don't know if you specified +set game ~/.ddaynormandy/dday as the game whether it would get the configuration there or not.

I don't know if spatulasnout has any suggestions for you.
